d3a_api_client/utils.py

```diff
@@ -271,3 +277,20 @@ def log_bid_offer_confirmation(message):
                          f"{round(energy, 2)} kWh at {price} cts/kWh")
     except Exception as e:
         logging.error(f"Logging bid/offer info failed.{e}")
+
+
+def get_simulation_config(simulation_id=simulation_id_from_env, domain_name=domain_name_from_env,
```

Comment:
I think that this method could benefit from some refactoring. You are accepting a `simulation_id` parameter, but in some cases it won't be used (when `SIMULATION_CONFIG_FILE_PATH` is available). This might lead to unexpected behaviour for the user.
Also, the `simulation_id_from_env` constant is `""` by default (as defined in `constants.API_CLIENT_SIMULATION_ID`) and I think this would cause exceptions later on.

Reply:
That was actually the motivation of this method to supersede the read values from `SIMULATION_CONFIG_FILE_PATH ` else take from the environment variable.

Reply:
I understand that the general behaviour is what you describe. But in general, when you pass an argument to a method, you expect that argument to be used (and not some default). Why not doing the opposite instead? If the user doesn't specify any argument, then you check the content of the `SIMULATION_CONFIG_FILE_PATH`?